docs/DeveloperGuide.md

```diff
@@ -6,33 +6,227 @@
+and each `TravelActivity` class object contains a date and duration. The `TravelAcivityList` also contains the following method:
+* `TravelActivityList#updateTravelActivity(travelActivityNumber, date, duration)`-- Updates the date and duration of the specified object of `TravelActivity` class type.
+
+Step 1. When the user launches the app for the first time, there will be no objects of type `TravelActivity` in `travelActivities`.
+Thus any attempts to call the command `update INDEX`
+will result in an output telling the user that no travel activity can be found.
+
+Step 2. After the user has added new travel activities into the `travelActivityList`, the user can execute
+`update 1 /date 2020-12-20 /duration 2hours` command to update the date and duration of the first travel activity
+in the travel activity list. The `update` command calls the `Parser#updateCommand(line, list)` which then calls the `TravelActivityList#updateTravelActivity(travelActivityNumber, date, duration)`
+. 
+
+Step 3. The method will then find the travel activity with the corresponding travel activity number in the `travelActivities` and then set the new date and duration
+of that travel activity.
+
+The sequence diagram below shows how an update operation goes through the parser component:
+![img_1.png](img_1.png)
+
+#[Proposed] Total Expense feature
+
+Proposed Implementation
+
+The proposed total expense feature is mainly carried out by `TravelActivityList`. It stores objects with class `TravelActivity`in an array called `travelActivities`
+and each `TravelActivity` class object contains an expense associated with it. The `TravelAcivityList` also contains the following method:
+* `TravelActivityList#totalExpense(type)`-- Returns the total expense of activities in the list with given type.
+
+Step 1. When the user launched the app for the first time, there will be no objects of type `TravelActivity` in `travelActivities`.
+Thus any attempts to call the command `total expense`
+will result in an output of `0`.
+
+Step 2. After the user has added new travel activities into the `travelActivityList`, the user can execute
+`totalexpense [/type TYPE]` command to get the total expense for the activities in the list. The `total expense` command calls the  `Parser#totalExpenseCommand(line, list)` which then calls the `TravelActivityList#totalExpense(type)`.
+
+Step 3. The method will loop through each activity in `travelActivities` and add up all the expenses for the given type. If no type is provided, it will default to TravelActivity type.
+
+![totalexpense.png](totalexpense.png)
+
+### Tag feature
+
+The tag feature is implemented through the tagCommand method, which allows users to add tags to specific travel activities stored in the system.
+
+The sequence diagram depicts the interaction between the user, the tagCommand method, the TravelActivityList instance, and the OmniException instance.
+
+![TagSequenceDiagram.png](TagSequenceDiagram.png)
```

Comment:
What are your thoughts on adopting a component-based architecture for diagrams to boost neatness, and ensuring the use of proper notation and conventions in sequence diagrams?